common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java

```diff
@@ -123,6 +139,84 @@ private static void findQuestForTracking(ContainerContent container, QuestInfo q
         }
     }
 
+    public static void scanDialogueHistory() {
+        ScriptedContainerQuery.QueryBuilder queryBuilder = ScriptedContainerQuery.builder(
+                        "Quest Book Dialogue History Query")
+                .onError(msg -> WynntilsMod.warn("Problem getting dialogue history in Quest Book: " + msg))
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(getQuestBookTitle(1))
+                .processContainer((c) -> {
+                    ItemStack dialogueHistoryItem = c.items().get(0);
+
+                    if (!ComponentUtils.getCoded(dialogueHistoryItem.getHoverName())
+                            .equals("§bDialogue History")) return;
+
+                    for (String line : ItemUtils.getLore(dialogueHistoryItem)) {
+                        Matcher matcher = DIALOGUE_HISTORY_PAGE_PATTERN.matcher(line);
+
+                        if (matcher.matches()) {
+                            int pageCount = Integer.parseInt(matcher.group(2));
+                            createActualDialogueHistoryQuery(pageCount);
```

Comment:
The scripted queries are really clumsy for this use case. I was planning to make it easier to create dynamic queries, but it turned out to be a hard problem defining an API with good usability to do dynamic querying, so I kind of got stuck. But this code might perhaps serve as a good example, so if I can do something that makes it better (so that we don't have to first check the number of pages, and *then* redo everything to iterate over them), I think we'd have a much better solution. 

For the time being though, this is a creative (and good-enough) solution.